gpytorch/test/base_keops_test_case.py

```diff
@@ -42,6 +42,11 @@ def test_forward_x1_eq_x2(self, ard=False, use_keops=True, **kwargs):
                 k2 = kern2(x1, x1).to_dense()
                 self.assertLess(torch.norm(k1 - k2), 1e-4)
 
+                # Test diagonal
+                d1 = kern1(x1, x1).diagonal(dim1=-1, dim2=-2)
+                d2 = kern2(x1, x1).diagonal(dim1=-1, dim2=-2)
+                self.assertLess(torch.norm(d1 - d2), 1e-4)
```

Comment:
This only tests that what the `diagonal` calls returns are the same, not that they actually compute the diagonal. You should add sth like this here.
```suggestion
                self.assertLess(torch.norm(d1 - d2), 1e-4)
                self.assertTrue(torch.equal(k1.diag(), d1))
```